File: ApiIntegration/app/src/main/java/com/example/apiintegration/MainActivity.java
package com.example.apiintegration;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    Button btn;
    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        btn = findViewById(R.id.button);
        tv = findViewById(R.id.text);
        btn.setOnClickListener(this);
    }

    public void getData(){
        String url = "https://catfact.ninja/fact";

        RequestQueue reqQue = Volley.newRequestQueue(this);
        StringRequest strReq = new StringRequest(Request.Method.GET, url,
        res->{

            tv.setText(res.toString());
        },
        err->            Toast.makeText(this, "Error", Toast.LENGTH_LONG).show());
        reqQue.add(strReq);

    }

    @Override
    public void onClick(View view) {
        getData();
    }
}

File: VechileMonitoringApp/app/src/main/java/com/example/vechilemonitoringapp/MainActivity.kt
package com.example.vechilemonitoringapp

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val btn1 = findViewById<Button>(R.id.register_vehicle)
        val btn2 = findViewById<Button>(R.id.check_health)

        btn1.setOnClickListener(){
            startActivity(Intent(this, RegisterActivity::class.java))
        }

        btn2.setOnClickListener(){
            startActivity(Intent(this, CheckHealthActivity::class.java ))
        }
    }
}